src/Resolver.hack

```diff
@@ -5,104 +5,46 @@ use namespace HH\Lib\{C, Vec};
 
 final class Resolver {
 
-    /**
-     * @see https://spec.graphql.org/draft/#sec-Response
-     */
-    const type TResponse = shape(
-        ?'data' => ?dict<string, mixed>, // missing data and null data are both valid states with different meanings
-        ?'errors' => vec<UserFacingError::TData>, // errors are optional but cannot be null (or empty) if present
-        ?'extensions' => dict<string, mixed>,
-    );
-
-    const type TOptions = shape(
-        // Enabling this causes errors to include information about the underlying error.
-        ?'verbose_errors' => bool,
-    );
-
-    public function __construct(private BaseSchema $schema, private this::TOptions $options = shape()) {}
+    public function __construct(private BaseSchema $schema) {}
 
     /**
-     * Operation name must be specified if the GraphQL request contains multiple operations.
-     *
-     * @see https://spec.graphql.org/draft/#sec-Execution
+     * Validate and resolve a GraphQL request against the schema..
      */
-    public async function resolve(
-        string $input,
-        dict<string, mixed> $variables = dict[],
-        ?string $operation_name = null,
-    ): Awaitable<this::TResponse> {
-        try {
-            $request = $this->parseRequest($input);
-        } catch (GraphQL\UserFacingError $e) {
-            return shape('errors' => $this->formatErrors(vec[$e]));
+    public async function resolve(IRequest $request): Awaitable<IResponse> {
+        $response = new Response($request);
+        // If the request is malformed, we can't proceed with execution.
+        if ($request is MalformedRequest) {
+            return $response->withErrors($request->getErrors());
         }
+        $request as WellformedRequest;
```

Comment:
Is there a nicer way to do this?

Reply:
Not that I know of :(

This kind of conditional branching can be avoided by putting the logic in the request class (so we have `$request->resolve()` instead of `Resolver::resolve($request)` but that's not always practical (also the "visitor" pattern addresses this more generically but that would be an even bigger overkill here probably).

Reply:
We'd need to update `resolve` to accept all the parameters one could use to instantiate a new resolver, in that case. I think I'll keep as-is for now and we can revisit. It would be nice to be able to write code like the following:

```hack
$request = GraphQL\Request::build($input, $variables);
$response = await $request->resolve();
```